File: app/src/main/java/com/example/back2me/auth/LoginActivity.java
package com.example.back2me.auth;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Toast;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import com.example.back2me.MainActivity;
import com.example.back2me.R;
import com.example.back2me.databinding.ActivityLoginBinding;
import com.google.android.material.textfield.TextInputEditText;
import com.google.firebase.auth.FirebaseAuth;

public class LoginActivity extends AppCompatActivity {

    private ActivityLoginBinding binding;
    private FirebaseAuth auth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityLoginBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        auth = FirebaseAuth.getInstance();

        setupClickListeners();
    }

    private void setupClickListeners() {
        // Back button
        binding.backButton.setOnClickListener(v -> finish());

        // Login button
        binding.buttonLogin.setOnClickListener(v -> attemptLogin());

        // Sign up link
        binding.textSignup.setOnClickListener(v -> {
            startActivity(new Intent(this, SignupActivity.class));
        });

        // Forgot password
        binding.textForgotPassword.setOnClickListener(v -> showForgotPasswordDialog());
    }

    private void attemptLogin() {
        String email = binding.inputEmail.getText().toString().trim();
        String password = binding.inputPassword.getText().toString().trim();

        // Validate
        if (TextUtils.isEmpty(email)) {
            binding.layoutEmail.setError("Email is required");
            return;
        }
        binding.layoutEmail.setError(null);

        if (TextUtils.isEmpty(password)) {
            binding.layoutPassword.setError("Password is required");
            return;
        }
        binding.layoutPassword.setError(null);

        // Show loading
        setLoading(true);

        // Attempt login
        auth.signInWithEmailAndPassword(email, password)
                .addOnSuccessListener(authResult -> {
                    setLoading(false);
                    Toast.makeText(this, "Login successful!", Toast.LENGTH_SHORT).show();

                    Intent intent = new Intent(this, MainActivity.class);
                    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
                    startActivity(intent);
                    finish();
                })
                .addOnFailureListener(e -> {
                    setLoading(false);
                    Toast.makeText(this, "Login failed: " + e.getMessage(), Toast.LENGTH_LONG).show();
                });
    }

    private void showForgotPasswordDialog() {
        View dialogView = getLayoutInflater().inflate(R.layout.dialog_forgot_password, null);
        TextInputEditText inputEmail = dialogView.findViewById(R.id.input_email);

        // Pre-fill with current email if entered
        String currentEmail = binding.inputEmail.getText().toString().trim();
        if (!currentEmail.isEmpty()) {
            inputEmail.setText(currentEmail);
        }

        new AlertDialog.Builder(this)
                .setTitle("Reset Password")
                .setMessage("Enter your email address and we'll send you a link to reset your password.")
                .setView(dialogView)
                .setPositiveButton("Send", (dialog, which) -> {
                    String email = inputEmail.getText().toString().trim();
                    if (!TextUtils.isEmpty(email)) {
                        sendPasswordResetEmail(email);
                    } else {
                        Toast.makeText(this, "Please enter your email", Toast.LENGTH_SHORT).show();
                    }
                })
                .setNegativeButton("Cancel", null)
                .show();
    }

    private void sendPasswordResetEmail(String email) {
        auth.sendPasswordResetEmail(email)
                .addOnSuccessListener(aVoid -> {
                    Toast.makeText(this,
                            "Password reset email sent to " + email,
                            Toast.LENGTH_LONG).show();
                })
                .addOnFailureListener(e -> {
                    Toast.makeText(this,
                            "Failed to send reset email: " + e.getMessage(),
                            Toast.LENGTH_LONG).show();
                });
    }

    private void setLoading(boolean loading) {
        binding.progressBar.setVisibility(loading ? View.VISIBLE : View.GONE);
        binding.buttonLogin.setEnabled(!loading);
        binding.buttonLogin.setText(loading ? "" : getString(R.string.sign_in));
    }
}

File: app/src/main/java/com/example/back2me/auth/LoginActivity.kt
// file: LoginActivity.kt (UPDATED)
package com.example.back2me.auth

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.back2me.MainActivity
import com.example.back2me.databinding.ActivityLoginBinding
import com.google.firebase.auth.FirebaseAuth
import com.example.back2me.AuthData // ✅ KEEP: Needed for setUserId

// ❌ REMOVE: import java.util.UUID (Not needed for static DB GUID or user ID logic)

class LoginActivity : AppCompatActivity() {

    private lateinit var binding: ActivityLoginBinding
    private lateinit var auth: FirebaseAuth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityLoginBinding.inflate(layoutInflater)
        setContentView(binding.root)

        auth = FirebaseAuth.getInstance()

        setupClickListeners()
    }

    private fun setupClickListeners() {
        binding.backButton.setOnClickListener {
            finish()
        }

        binding.buttonLogin.setOnClickListener {
            val email = binding.inputEmail.text.toString().trim()
            val password = binding.inputPassword.text.toString().trim()

            if (validateInput(email, password)) {
                loginUser(email, password)
            }
        }

        binding.textSignup.setOnClickListener {
            startActivity(Intent(this, SignupActivity::class.java))
        }

        // Social login buttons (not implemented)
        binding.buttonGoogle.setOnClickListener {
            Toast.makeText(this, "Google login not implemented", Toast.LENGTH_SHORT).show()
        }
        binding.buttonFacebook.setOnClickListener {
            Toast.makeText(this, "Facebook login not implemented", Toast.LENGTH_SHORT).show()
        }
        binding.buttonTwitter.setOnClickListener {
            Toast.makeText(this, "Twitter login not implemented", Toast.LENGTH_SHORT).show()
        }
    }

    private fun validateInput(email: String, password: String): Boolean {
        if (email.isEmpty()) {
            binding.inputEmail.error = "Please enter your email"
            binding.inputEmail.requestFocus()
            return false
        }

        // Basic email validation
        if (!android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
            binding.inputEmail.error = "Please enter a valid email"
            binding.inputEmail.requestFocus()
            return false
        }

        if (password.isEmpty()) {
            binding.inputPassword.error = "Please enter your password"
            binding.inputPassword.requestFocus()
            return false
        }

        if (password.length < 6) {
            binding.inputPassword.error = "Password must be at least 6 characters"
            binding.inputPassword.requestFocus()
            return false
        }

        return true
    }

    private fun loginUser(email: String, password: String) {
        showLoading(true)

        auth.signInWithEmailAndPassword(email, password)
            .addOnCompleteListener(this) { task ->
                showLoading(false)

                if (task.isSuccessful) {

                    // 💥 FIX: The DB GUID is static. We must save the authenticated User ID.
                    val userId = task.result?.user?.uid
                    if (userId != null) {
                        AuthData.setUserId(userId)
                    }

                    Toast.makeText(this, "Login successful!", Toast.LENGTH_SHORT).show()
                    navigateToMain()
                } else {
                    val errorMessage = task.exception?.message ?: "Login failed"
                    Toast.makeText(this, errorMessage, Toast.LENGTH_LONG).show()
                }
            }
    }

    private fun showLoading(show: Boolean) {
        binding.progressBar.visibility = if (show) View.VISIBLE else View.GONE
        binding.buttonLogin.isEnabled = !show
    }

    private fun navigateToMain() {
        val intent = Intent(this, MainActivity::class.java)
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
        startActivity(intent)
        finish()
    }
}